substrate/frame/staking/src/lib.rs

```diff
@@ -387,7 +388,97 @@ impl<AccountId: Ord> Default for EraRewardPoints<AccountId> {
 	}
 }
 
+/// The payout destination for an account.
+///
+/// NOTE: Being lazily migrated to. Logic pertaining to this enum has been introduced to `set_payee`
+/// and payout logic, replacing `RewardDestination`.
+#[derive(PartialEq, Eq, Copy, Clone, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub enum PayoutDestination<AccountId> {
+	/// Payout goes into the stash account and is added to bond.
+	Stake,
+	/// Deposit the specified percentage of payout to the specified account as free balance, and
+	/// pay the rest into the stash account and add to bond. 0% and 100% should be disallowed and
+	/// handled as `Stake` and `Deposit` respectively.
+	Split((Perbill, AccountId)),
+	/// Deposit payout as free balance to an account.
+	Deposit(AccountId),
+	/// Receive no payout.
+	Forgo,
+}
+
+// Used for testing and benchmarking where stash and controller accounts are sometimes generated
+// after providing the desired `PayoutDestination`. In such scenarios the provided `Alias` variant
+// is used. If payout destination accounts are already known, `Direct` can be used.
+#[derive(PartialEq, Copy, Clone)]
+pub enum PayoutRoute<AccountId> {
```

Comment:
make two impl blocks, move `from_route` in there, mark it as `#[cfg(any(test, feature = "runtime-benchmarks"))]`.